src/dbnode/storage/index/aggregate_results.go

```diff
@@ -42,10 +45,44 @@ type aggregatedResults struct {
 	idPool    ident.Pool
 	bytesPool pool.CheckedBytesPool
 
-	pool       AggregateResultsPool
-	valuesPool AggregateValuesPool
-
+	pool             AggregateResultsPool
+	valuesPool       AggregateValuesPool
 	encodedDocReader docs.EncodedDocumentReader
+
+	iOpts   instrument.Options
+	metrics usageMetrics
+}
+
+type usageMetrics struct {
+	total tally.Counter
+
+	totalTerms   tally.Counter
+	dedupedTerms tally.Counter
+
+	totalFields   tally.Counter
+	dedupedFields tally.Counter
+}
+
+func newUsageMetrics(ns ident.ID, iOpts instrument.Options) usageMetrics {
```

Comment:
do we need to reset this with every aggregate result? Is the namespace important enough?